src/Compiler/Service/FSharpCheckerResults.fs

```diff
@@ -1007,6 +1053,49 @@ type internal TypeCheckInfo
                     Some (List.map ItemWithNoInst items, denv, m)
                     |> Option.map toCompletionItems
 
+            // Completion at ' match x with S... -> () ' unless we're dotting into a type/module/...
+            | Some (CompletionContext.Match (MatchContext.ClausePatternIdentifier range)) when origLongIdentOpt.IsNone || origLongIdentOpt.Value.IsEmpty ->
+                match GetUnionCasesAndActivePatternsEnvironmentLookupResolutionsAtPosition range.End with
+                | Some (isUnionInScopeAsUnqualified, items, denv, range) ->
+                    let items =
+                        items
+                        |> List.map (fun item ->
+                            let unresolved =
+                                match item with
+                                | Item.UnionCase (uci, requiresQualifiedAccess) when not isUnionInScopeAsUnqualified ->
+                                    Some {
+                                        FullName = uci.Tycon.CompiledRepresentationForNamedType.FullName
+                                        Namespace = trimPathByDisplayEnvList denv uci.Tycon.CompilationPath.DemangledPath |> List.toArray
+                                        DisplayName = if requiresQualifiedAccess then $"{uci.Tycon.DisplayName}.{uci.DisplayName}" else uci.DisplayName
+                                    }
+                                | _ -> None
+
+                            {
+                                ItemWithInst = ItemWithNoInst item
+                                MinorPriority = 0
+                                Kind = CompletionItemKind.Other
+                                IsOwnMember = false
+                                Type = None
+                                Unresolved = unresolved
```

Comment:
We should add `///` comments to all of `CompletionItem` fields, it's a little unclear what `Unresolved` is about. Could you explain here?  I've forgotten the details.

Reply:
I'm not sure about all the record fields, but having a `Some` value in `Unresolved` will cause that completion item to open a namespace/module on use.